pkg/app/piped/apistore/eventstore/store.go

```diff
@@ -177,3 +172,31 @@ func (s *store) GetLatest(ctx context.Context, name string, labels map[string]st
 	s.latestEvents[key] = resp.Event
 	return resp.Event, true
 }
+
+func (s *store) UpdateStatuses(ctx context.Context, latestEvents []model.Event) error {
+	es := make([]*pipedservice.ReportEventStatusesRequest_Event, 0, len(latestEvents))
+	for _, e := range latestEvents {
+		es = append(es, &pipedservice.ReportEventStatusesRequest_Event{
+			Id:                e.Id,
+			Status:            e.Status,
+			StatusDescription: e.StatusDescription,
+		})
+	}
+	if _, err := s.apiClient.ReportEventStatuses(ctx, &pipedservice.ReportEventStatusesRequest{Events: es}); err != nil {
+		return fmt.Errorf("failed to report event statuses: %w", err)
+	}
+
+	// Update cached events.
+	for _, e := range latestEvents {
+		s.mu.Lock()
+		cached, ok := s.latestEvents[e.EventKey]
+		if ok && cached.Id != e.Id {
+			// There is already an event newer than the given one.
+			s.mu.Unlock()
+			continue
+		}
+		s.latestEvents[e.EventKey] = &e
```

Comment:
Let's update just the `status` field as what the function name is saying instead of the whole object.

Reply:
For sure!